packages/repository/src/repositories/relation.repository.ts

```diff
@@ -53,10 +97,45 @@ export class DefaultHasManyEntityCrudRepository<
     );
   }
 
-  async find(filter?: Filter | undefined, options?: Options): Promise<T[]> {
+  async find(filter?: Filter, options?: Options): Promise<T[]> {
     return await this.targetRepository.find(
       constrainFilter(filter, this.constraint),
       options,
     );
   }
+
+  async update(entity: DataObject<T>, options?: Options): Promise<boolean> {
+    return await this.targetRepository.update(entity, options);
```

Comment:
I think you forgot to apply `constrainFilter`?

It would be great if you could work on this in a test-first/test-driven way - start with a failing unit or integration test and only then write an implementation.

Reply:
🙈 Sorry I should have marked this PR as not ready for review

> t would be great if you could work on this in a test-first/test-driven way - start with a failing unit or integration test and only then write an implementation.

True 👍 The missing function is `constrainDataObject` IIUC